docs/serving/samples/multi-container/README.md

```diff
@@ -0,0 +1,274 @@
+```
+
+## Before you begin
+
+- A Kubernetes cluster with Knative installed and DNS configured. Follow the
+  [installation instructions](../../../install/README.md) if you need to
+  create one.
+- [Docker](https://www.docker.com) installed and running on your local machine,
+  and a Docker Hub account configured (we'll use it for a container registry).
+- Make sure multi-container flag is enabled as part of `config-features` configmap.
+
+## Recreating the sample code
+
+Multi container has more than one container, For testing, we will create two containers, the serving container and sidecar container.
+### ServingContainer
+1. `cd knative-docs/docs/serving/samples/multi-container/servingcontainer`
+1. Create a new file named `servingcontainer.go` and paste the following code. This
+   code creates a basic web server which listens on port 8881:
+
+   ```go
+   package main
+   
+   import (
+   	"fmt"
+   	"io/ioutil"
+   	"log"
+   	"net/http"
+   )
+   
+   func handler(w http.ResponseWriter, r *http.Request) {
+   	log.Println("serving container received a request.")
+   	res, err := http.Get("http://127.0.0.1:8882")
+   	if err != nil {
+   		log.Fatal(err)
+   	}
+   	resp, err := ioutil.ReadAll(res.Body)
+   	if err != nil {
+   		log.Fatal(err)
+   	}
+   	fmt.Fprintln(w, string(resp))
+   }
+   
+   func main() {
+   	log.Print("serving container started...")
+   	http.HandleFunc("/", handler)
+   	log.Fatal(http.ListenAndServe(":8881", nil))
+   }
+   ```
+
+1. In your project directory, create a file named `Dockerfile` and copy the code
+   block below into it. For detailed instructions on dockerizing a Go app, see
+   [Deploying Go servers with Docker](https://blog.golang.org/docker).
+
+   ```docker
+   # Use the official Golang image to create a build artifact.
+   # This is based on Debian and sets the GOPATH to /go.
+   # https://hub.docker.com/_/golang
+   FROM golang:1.13 as builder
+   
+   # Create and change to the app directory.
+   WORKDIR /app
+   
+   # Retrieve application dependencies using go modules.
+   # Allows container builds to reuse downloaded dependencies.
+   COPY go.* ./
+   RUN go mod download
+   
+   # Copy local code to the container image.
+   COPY . ./
+   
+   # Build the binary.
+   # -mod=readonly ensures immutable go.mod and go.sum in container builds.
+   RUN CGO_ENABLED=0 GOOS=linux go build -mod=readonly -v -o servingcontainer
+   
+   # Use the official Alpine image for a lean production container.
+   # https://hub.docker.com/_/alpine
+   # https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds
+   FROM alpine:3
+   RUN apk add --no-cache ca-certificates
+   
+   # Copy the binary to the production image from the builder stage.
+   COPY --from=builder /app/servingcontainer /servingcontainer
+   
+   # Run the web service on container startup.
+   CMD ["/servingcontainer"]
+   ```
+
+### SidecarContainer
+1. ```shell
```

Comment:
Same comment as above - use conistent formatting for code blocks and explain what the command does, e.g. "1. Navigate to...."